class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+            $action['type']           = 'fail_payment';
+            $action['transaction_id'] = $attributes['razorpay_payment_id'];
+            $action['entry_id']       = $entry['id'];
+            $action['amount']         = $entry['payment_amount'];
+        }
+
+        return $action;
+    }
+
+    public function post_callback($callback_action, $callback_result) {
+        $entry               = GFAPI::get_entry($callback_action['entry_id']);
+        $feed                = $this->get_payment_feed( $entry );
+        $transaction_id      = rgar($callback_action, 'transaction_id');
+        $amount              = rgar($callback_action, 'amount');
+
+        do_action('gform_razorpay_post_payment_' . $callback_action['type'],  $_POST, $entry, $feed);
+    }
+
+    public function generate_razorpay_form($entry, $customer_fields, $form)
+    {
+        $feed = $this->get_payment_feed($entry, $form);
+
+        $razorpay_args = array(
+            'key'         => $this->get_plugin_setting(self::GF_RAZORPAY_KEY),
+            'name'        => $form['name'],
+            'amount'      => (int) round($entry['payment_amount']*100),
+            'currency'    => $entry['currency'],
+            'description' => $form['description'],
+            'prefill'     => array(
+                'name'    => $customer_fields['billingInformation_firstName'],
+                'email'   => $customer_fields['billingInformation_email'],
+            ),
+            'notes'       => array(
+                'gravity_forms_order_id' => $entry['id']
+            ),
+            'order_id'    => $entry['razorpay_order_id'],
+        );
+
+        $json = $razorpay_args;
+
+        $redirect_url = '?page=gf_razorpay_callback';
+
+
+        wp_enqueue_script('razorpay-script', plugin_dir_url( __FILE__ ) . 'script.js',
+            array('checkout'));
+
+        wp_localize_script('razorpay-script', 'razorpay_script_vars', array(
+            'data' => $json
+        ));
+
+        wp_register_script( 'checkout', 'https://checkout.razorpay.com/v1/checkout.js', null, null);
+
+        wp_enqueue_script('checkout');
+
+        return $this->generate_order_form($redirect_url);
+    }
+
+    function generate_order_form($redirect_url)
+    {
+        $html = <<<EOT
+<form id ='razorpayform' name='razorpayform' action="$redirect_url" method='POST'>
+    <input type='hidden' name='razorpay_payment_id' id='razorpay_payment_id'>
+    <input type='hidden' name='razorpay_signature'  id='razorpay_signature' >
+</form>
+<p id='msg-razorpay-success'  style='display:none'>
+    Please wait while we are processing your payment.
+</p>
+<p>
+    <button id='btn-razorpay'>Pay Now</button>
+    <button id='btn-razorpay-cancel' onclick='document.razorpayform.submit()'>Cancel</button>
+</p>
+EOT;
+        return $html;
+    }
+
+    public function is_callback_valid()
+    {
+        if (rgget( 'page' ) !== 'gf_razorpay_callback')
```

Comment:
Add a comment explaining what this does. Similarly, add comments for all ambiguous methods.